prow/cmd/build/controller.go

```diff
@@ -307,6 +345,20 @@ func (c *controller) buildID(pj prowjobv1.ProwJob) (string, string, error) {
 	return id, url, nil
 }
 
+func (c *controller) terminateDupProwJobs(ctx string, namespace string) error {
+	jobs, err := c.getProwJobs(namespace)
+	if err != nil {
+		return err
+	}
+	return c.aborter.TerminateOlderPresubmitJobs(jobs, func(toCancel prowjobv1.ProwJob) {
```

Comment:
Is there a reason this needs a closure? I'd think we could encapsulate this as something like:

```golang
currentJobs, olderJobs := filterJobs(jobs)
for _, j := range olderJobs {
  c.deleteBuild(ctx, namesapce, j.GetName())
}
```

Reply:
The resources `cleanup` closure is called before marking the prow job as aborted. Both, the logic to detect the older prow jobs as well as to update the aborted state of the prow job is encapsulated in the `TerminateOlderPresubmitJobs` method.